src/m365/spo/commands/tenant/tenant-site-archive.spec.ts

```diff
@@ -0,0 +1,185 @@
+    auth.connection.spoUrl = 'https://contoso.sharepoint.com';
+    commandInfo = cli.getCommandInfo(command);
+  });
+
+  beforeEach(() => {
+    log = [];
+    logger = {
+      log: async (msg: string) => {
+        log.push(msg);
+      },
+      logRaw: async (msg: string) => {
+        log.push(msg);
+      },
+      logToStderr: async (msg: string) => {
+        log.push(msg);
+      }
+    };
+
+    sinon.stub(cli, 'promptForConfirmation').callsFake(async () => {
+      promptIssued = true;
+      return false;
+    });
+
+    promptIssued = false;
+  });
+
+  afterEach(() => {
+    sinonUtil.restore([
+      request.post,
+      cli.promptForConfirmation
+    ]);
+  });
+
+  after(() => {
+    sinon.restore();
+    auth.connection.active = false;
+    auth.connection.spoUrl = undefined;
+  });
+
+  it('has correct name', () => {
+    assert.strictEqual(command.name, commands.TENANT_SITE_ARCHIVE);
+  });
+
+  it('has a description', () => {
+    assert.notStrictEqual(command.description, null);
+  });
+
+  it('fails validation if url is not a valid SharePoint URL', async () => {
+    const actual = await command.validate({ options: { url: 'abc' } }, commandInfo);
+    assert.notStrictEqual(actual, true);
+  });
+
+  it('passes validation when valid url passed', async () => {
+    const actual = await command.validate({ options: { url: 'https://contoso.sharepoint.com/sites/Sales' } }, commandInfo);
+    assert.strictEqual(actual, true);
+  });
+
+  it('prompts before archiving the site when force option not passed', async () => {
+    await command.action(logger, { options: { url: 'https://contoso.sharepoint.com/sites/Sales' } });
+
+    assert(promptIssued);
+  });
+
+  it('aborts archiving the site when prompt not confirmed', async () => {
+    const postStub = sinon.stub(request, 'post').resolves();
+    await command.action(logger, { options: { url: 'https://contoso.sharepoint.com/sites/Sales' } });
+    assert(postStub.notCalled);
+  });
+
+  it('archive the site when prompt confirmed', async () => {
+    const postStub = sinon.stub(request, 'post').callsFake(async (opts) => {
+      if (opts.url === `https://contoso-admin.sharepoint.com/_vti_bin/client.svc/ProcessQuery` &&
+        opts.data === `<Request AddExpandoFieldTypeSuffix="true" SchemaVersion="15.0.0.0" LibraryVersion="16.0.0.0" ApplicationName="${config.applicationName}" xmlns="http://schemas.microsoft.com/sharepoint/clientquery/2009">\n<Actions>\n  <ObjectPath Id="2" ObjectPathId="1" />\n  <ObjectPath Id="4" ObjectPathId="3" />\n  <Query Id="5" ObjectPathId="3">\n    <Query SelectAllProperties="true">\n      <Properties />\n    </Query>\n  </Query>\n</Actions>\n<ObjectPaths>\n  <Constructor Id="1" TypeId="{268004ae-ef6b-4e9b-8425-127220d84719}" />\n  <Method Id="3" ParentId="1" Name="ArchiveSiteByUrl">\n    <Parameters>\n      <Parameter Type="String">https://contoso.sharepoint.com/sites/Sales</Parameter>\n    </Parameters>\n  </Method>\n</ObjectPaths>\n</Request>`) {
```

Comment:
Instead of checking the body like this, let's put it in an assert. This way, the error will be much clearer when the test fails.